chapter3.adoc

```diff
@@ -16,42 +16,20 @@ Will be addressed as follows.
-| allsecured |  The field is 1 when all currently selected hart are secured and not running in debuggable privilege levels | R    | 0    
-| anysecured |  The field is 1 when any currently selected hart is secured and not running in debuggable privilege levels  | R    | 0    
-|===============================================================================================================================================
-
 === Halt
 
 The hart can only be halted when it is running in privilege levels specified in <<dbgpriv>>. Otherwise, the halt request will remain pending. Consequently, the response to the external debugger request may take longer than one second, and this constraint must be eliminated.
 
-When machine mode is not permitted (`mdbgen` set to 0) to engage in debugging, the halt-on-reset (resethaltreq) operation must fail and set cmderr to 6 (security fault error).
 
 === Reset
 
-The reset operations must be safeguarded against various attacks. The RISC-V Debug Specification defines the hartreset and ndmreset operations, which reset selected harts or the entire system (excluding the Debug Module), respectively. The hartreset operation must be prohibited when machine mode is not allowed to be debugged. The security fault error will be raised if the operation is issued when `mdbgen` is 0. 
+The reset operations must be safeguarded against various attacks. The RISC-V Debug Specification defines the hartreset and haltresetreq operations, which reset selected harts. The reset operations must be prohibited when machine mode is not allowed to be debugged. The security fault error will be raised if the operation is issued when mdbgen[i] is 0. The debugger could check the error by polling allsecfault/anysecfault bits in dmstatus as specified in <<regdmstatus>>. 
 
-The ndmreset operation is a system-level reset not tied to hart privilege levels and can only be secured by the system. Thus, it must be de-featured. Issuing ndmreset operation results in a security fault error when `mdbgen` is 0. 
+The ndmreset operation is a system-level reset not tied to hart privilege levels and reset the entire system (excluding the Debug Module). It can only be secured by the system. Thus, it must be de-featured. The debugger can determine support for the ndmreset operation by setting the field to 1 and subsequently verifying the returned value upon reading.
 
 === Keepalive
 
-The keepalive operation serves as an optional request for the hart to remain available for debugger. It is only allowed when machine mode is permitted to debug. Otherwise, it causes a security fault error.  
+The keepalive operation serves as an optional request for the hart to remain available for debugger. It is only allowed when machine mode is permitted to debug. Otherwise, it causes a security fault error. The debugger could monitor the error by polling allsecfault/anysecfault in dmstatus as described in <<regdmstatus>>. 
```

Comment:
I think it is better to use similar wording with hartreset and haltresetreq: The security fault error will be raised if the operation is issued when mdbgen[i] is 0

Reply:
Sure, will update